### PR TITLE
Add xml docs

### PR DESCRIPTION
**Pull request by sungam3r**

### CODE REVIEW

File: src/GraphQLParser/Location.cs [hunk 1]
```diff
@@ -2,10 +2,18 @@
 
 namespace GraphQLParser
 {
+    /// <summary>
+    /// Provides the ability to decode a linear character position into a row and column number.
```
Comment (sungam3r) [inline src/GraphQLParser/Location.cs:6]:
```suggestion
    /// Provides the ability to decode a linear character position into a line and column number.
```

File: src/GraphQLParser/AST/ASTNodeKind.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        Document,
+        OperationDefinition,
+        VariableDefinition,
+        Variable,
+        SelectionSet,
+        Field,
+        Argument,
+        FragmentSpread,
+        InlineFragment,
+        FragmentDefinition,
```
Comment (Shane32) [inline src/GraphQLParser/AST/ASTNodeKind.cs:23]: Applying a comment to `Name` does not automatically apply it to the other enumeration entries.

**Review by Shane32 — APPROVED** [review]
Looks like a good start!  Any idea how many more warnings are left?

File: src/GraphQLParser/AST/ASTNodeKind.cs (hunk 2 above)
Comment (Shane32) [reply src/GraphQLParser/AST/ASTNodeKind.cs:23]: (does it?)

Comment (sungam3r) [reply src/GraphQLParser/AST/ASTNodeKind.cs:23]: Of course I added comment above only for `Name`.

**Comment by sungam3r** [comment]
I'm not a native speaker. It's better to double-check what I wrote in the comments.

**Comment by Shane32** [comment]
I read all the comments.  They seem fine.

**Pull request merged**